tfhe/src/strings/server_key/mod.rs

```diff
@@ -203,9 +203,8 @@ impl ServerKey {
         &self,
         condition: &BooleanBlock,
         mut true_ct: FheString,
-        false_ct: &FheString,
+        mut false_ct: FheString,
     ) -> FheString {
```

Comment:
I  would take both references and clone both strings in the function, this is just a foot gun waiting to happen, if you want to avoid the FheString empty thing, take options of references or a custom enum to manage the empty string case maybe